sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -253,28 +258,44 @@ public static boolean hasCrashedInLastSession() {
 
     /**
      * Provides information about any available crash report from the last session, if it crashed.
-     * It also returns {@code null} if Crashes service failed to load the crash information.
      *
-     * @return The crash report from the last session if one was set.
+     * @param listener The listener that will receive crash in the last session.
      * @see #hasCrashedInLastSession()
      */
-    @Nullable
-    public static ErrorReport getLastSessionCrashReport() {
-        return getInstance().getInstanceLastSessionCrashReport();
+    public static void getLastSessionCrashReport(LastCrashErrorReportListener listener) {
+        getInstance().getInstanceLastSessionCrashReport(listener);
     }
 
     /**
      * Implements {@link #hasCrashedInLastSession()} at instance level.
      */
     private synchronized boolean hasInstanceCrashedInLastSession() {
-        return mFoundLastSessionErrorReport;
+        return mFoundLastSessionErrorReport != LastCrashProcessingStatus.NOT_FOUND &&
+                mFoundLastSessionErrorReport != LastCrashProcessingStatus.PREPARE_PROCESSING;
     }
 
     /**
-     * Implements {@link #getLastSessionCrashReport()} at instance level.
+     * Implements {@link #getLastSessionCrashReport(LastCrashErrorReportListener)} at instance level.
      */
-    private synchronized ErrorReport getInstanceLastSessionCrashReport() {
-        return mLastSessionErrorReport;
+    private synchronized void getInstanceLastSessionCrashReport(LastCrashErrorReportListener listener) {
+        switch (mFoundLastSessionErrorReport) {
+            case PREPARE_PROCESSING:
+            case PROCESSING:
+                if (mLastCrashErrorReportListeners == null) {
+                    mLastCrashErrorReportListeners = new ArrayList<>();
+                }
+                mLastCrashErrorReportListeners.add(listener);
+                break;
+            case PROCESSED:
+                listener.onSuccess(mLastSessionErrorReport);
+                break;
+            case PROCESSING_FAILED:
+                listener.onFailure();
```

Comment:
We should not have a failure case for getSessionErrorReport, either return result or return null like before. The sync version has to return either a result or null and never throw an exception, so why would the async version introduce that?